lib/matplotlib/collections.py

```diff
@@ -892,15 +928,21 @@ class PathCollection(_CollectionWithSizes):
     The most basic `Collection` subclass, created e.g. by `~.Axes.scatter`.
     """
 
-    @docstring.dedent_interpd
     def __init__(self, paths, sizes=None, **kwargs):
         """
-        *paths* is a sequence of `matplotlib.path.Path` instances.
-
-        %(Collection)s
+        Parameters
+        ----------
+        paths : Sequence[matplotlib.path.Path]
+            The paths that will make up the `Collection`.
+        sizes : Sequence[float]
```

Comment:
underscore, I think https://has2k1.github.io/param-type-demo/numpy-param-types.html

Reply:
Okay so it's definitely underscore, but I see that we also definitely use dash...

```bash
$ grep -iIR array-like . 2>/dev/null | wc -l
200
$ grep -iIR array_like . 2>/dev/null | wc -l
7
```

I'll go with the one we're already using. Thanks for the heads up!

Reply:
We are currently using `array-like`. See https://github.com/matplotlib/matplotlib/issues/10225#issuecomment-377995108 and https://matplotlib.org/devdocs/devel/documenting_mpl.html#parameter-type-descriptions

If there are compelling reasons to change that, it should be a separate PR as it will need discussion and affects a lot of docstrings.

Reply:
Sorry, to clarify, I'll definitely use `array-like`.

I think there's no compelling reason (besides matching the rest of the scientific python ecosystem, which all use array_like), I'll have to open a separate PR to fix the examples that I did find of `array_like`.

I assume back when you checked in 2018 there still wasn't consensus, but after looking threw several dozen pages in pandas, scipy, numpy, seaborn, etc., everybody uses `array_like` for the numpydoc type description and "array-like" in english sentences describing the property.